src/renderer/main/grid-layout/grid-modules/underlays/ActiveChanges.svelte

```diff
@@ -10,32 +12,29 @@
 
   const dispatch = createEventDispatcher();
 
-  let [dx, dy] = [device?.dx, device?.dy];
-
   let isChanged = false;
-  $: {
-    try {
-      const events = $runtime
-        .find((e) => e.dx == dx && e.dy == dy)
-        .pages[$user_input.event.pagenumber].control_elements.find(
-          (e) => e.controlElementNumber == elementNumber
-        ).events;
 
+  $: handleDeviceChange(device);
+
+  function handleDeviceChange(rt) {
+    const { dx, dy } = device;
+    const ui = get(user_input);
+    try {
```

Comment:
Eliminate try ... catch using proper conditional statements. This will greatly improve readability of the code!